contracts/DaoModule.sol

```diff
@@ -111,13 +131,9 @@ contract DaoModule {
         if (nonce > 0) {
             // Previous nonce must have been invalidated by the oracle.
             // However, if the proposal was internally invalidated, it should not be possible to ask it again.
-            bytes32 invalidatedQuestionId = getQuestionId(
-                templateId, question, arbitrator, timeout, 0, nonce - 1
-            );
             bytes32 currentQuestionId = questionIds[questionHash];
             require(currentQuestionId != INVALIDATED, "This proposal has been marked as invalid");
-            require(currentQuestionId == invalidatedQuestionId, "Unexpected existing question id for this proposal");
-            require(oracle.resultFor(invalidatedQuestionId) == INVALIDATED, "Previous proposal was not invalidated");
+            require(oracle.resultFor(currentQuestionId) == INVALIDATED, "Previous proposal was not invalidated");
```

Comment:
The case `nonce > 0` for a new question hash is ok because the question id 0 can never be resolved. Would it make sense to remove the `require(currentQuestionId != INVALIDATED)` for the same reason, since `INVALIDATED` can never be resolved?

Also a note, since it took a while to check: the case of reusing the same nonce twice when calling `addProposalWithNonce` is ok because `oracle.askQuestion` can only be called once for the same parameters.

Reply:
I left in the additional checks as they cost very little gas